test/sql/upsert/upsert_unique_null.test

```diff
@@ -0,0 +1,45 @@
+# name: test/sql/upsert/upsert_unique_null.test
+# group: [upsert]
+
+statement ok
+pragma enable_verification;
```

Comment:
Can we perhaps include [the Postgres INSERT ON CONFLICT tests](https://github.com/postgres/postgres/blob/master/src/test/regress/sql/insert_conflict.sql#L4)? (note not everything in that file needs to work - there might be some syntax/features we don't support in there, like `partition by`)

Reply:
I have had to adapt the tests quite a bit, as there are a couple of features that they use extensively throughout the test that we don't support:

ON CONFLICT checking on non-implicit Indexes, they also drop and create new indexes a lot throughout the tests, which would have to be a ALTER TABLE in our case, but we don't support adding/dropping constraints after creation.

Their indexes can be a lot more complex, containing among others function expressions and the option to add a WHERE clause.

Altering the columns that are indexed on.